backend/index.php

```diff
@@ -81,6 +65,139 @@ public function query()
+            $diff = [];
+            foreach ($decodedResult['hits']['hits'] as $row) {
+                $diff[] = $this->parseSingleInfo($row['_source']);
+            }
+
+            $this->printResponse($this->getSystemDiff($diff[0], $diff[1]));
+        }
+
+        $this->printResponse(['message' => 'Requested rows not found'], self::STATUS_ERROR, 404);
+    }
+
+    private function getSystemDiff($rowFirst, $rowSecond)
+    {
+        $rowFirstFileName  = DIRECTORY_SEPARATOR.'tmp'.DIRECTORY_SEPARATOR.$rowFirst['engine'];
+        $rowSecondFileName = DIRECTORY_SEPARATOR.'tmp'.DIRECTORY_SEPARATOR.$rowSecond['engine'];
+        unset($rowFirst['engine'], $rowSecond['engine']);
+        $keys = array_keys($rowFirst);
+
+        $compare = [];
+        foreach ($keys as $key) {
+            if (is_array($rowFirst[$key])) {
+                $rowFirst[$key] = $this->formatArrayToText($rowFirst[$key]);
+            }
+
+            if (is_array($rowSecond[$key])) {
+                $rowSecond[$key] = $this->formatArrayToText($rowSecond[$key]);
+            }
+
+            file_put_contents($rowFirstFileName, $rowFirst[$key]."\n");
+            file_put_contents($rowSecondFileName, $rowSecond[$key]."\n");
+
+
+            $command = 'diff -U 1000000 -u '.$rowFirstFileName.' '.$rowSecondFileName;
+            $output  = [];
+            exec($command, $output, $exitCode);
+            if ($exitCode <= 1) {
+                if ($output !== []) {
+                    $compare[$key] = implode("\n", $output);
+                } else {
+                    $compare[$key]
+                        = "diff --git $rowFirstFileName $rowFirstFileName\n--- $rowFirstFileName\n+++ $rowSecondFileName\nNo diff\n"
+                        .$rowFirst[$key]."+".$rowFirst[$key];
+                }
+            } else {
+                $this->printResponse(['message' => 'Error during diff generation'], self::STATUS_ERROR, 400);
+            }
+        }
+
+        return $compare;
+    }
+
+    private function formatArrayToText($array)
+    {
+        $formatted = '';
+        foreach ($array as $k => $v) {
+            $formatted .= "$k: $v\n";
+        }
+
+        return $formatted;
+    }
+
+    private function parseSingleInfo(array $row)
+    {
+        return [
+            'query'       => [
+                'original_query' => $row['original_query'],
+                'modified_query' => $row['modified_query'],
+            ],
+            'performance' => [
+                'cold'        => $row['cold'],
+                'fastest'     => $row['fastest'],
+                'slowest'     => $row['slowest'],
+                'warmup_time' => $row['warmup_time'],
+                'avg'         => $row['avg'],
+                'cv'          => $row['cv'],
+
+                'avg_fastest' => $row['avg_fastest'],
+
+                'cv_avg_fastest' => $row['cv_avg_fastest'],
+                'times_count'    => count($row['times']),
```

Comment:
```suggestion
                'Avg(80% fastest)' => $row['avg_fastest'],

                'CV of avg(80% fastest)' => $row['cv_avg_fastest'],
                'Times count'    => count($row['times']),
```

Reply:
Let's call it not `Times count`, but `Query runs`. It should be more clear then since "times" is ambiguous.